graphql-bal-service/src/main/java/io/ballerina/graphql/schema/utils/Utils.java

```diff
@@ -0,0 +1,340 @@
+    }
+
+    /**
+     * Get encoded schema string from the given node.
+     */
+    public static String getSchemaString(ServiceDeclarationNode node) throws SchemaGenerationException {
+        if (node.metadata().isPresent()) {
+            if (!node.metadata().get().annotations().isEmpty()) {
+                MappingConstructorExpressionNode annotationValue = getAnnotationValue(node.metadata().get());
+                return getSchemaStringFieldFromValue(annotationValue);
+            }
+        }
+        throw new SchemaGenerationException(DiagnosticMessages.SDL_SCHEMA_102, null, MESSAGE_MISSING_ANNOTATION);
+    }
+
+    /**
+     * Get encoded schema string from the given node.
+     */
+    public static String getSchemaString(ObjectConstructorExpressionNode node) throws SchemaGenerationException {
+        if (!node.annotations().isEmpty()) {
+            for (AnnotationNode annotationNode: node.annotations()) {
+                if (isGraphqlServiceConfig(annotationNode) && annotationNode.annotValue().isPresent()) {
+                    return getSchemaStringFieldFromValue(annotationNode.annotValue().get());
+                }
+            }
+        }
+        throw new SchemaGenerationException(DiagnosticMessages.SDL_SCHEMA_102, null, MESSAGE_MISSING_ANNOTATION);
+    }
+
+    /**
+     * Get annotation value string from the given metadata node.
+     */
+    private static MappingConstructorExpressionNode getAnnotationValue(MetadataNode metadataNode)
+            throws SchemaGenerationException {
+        for (AnnotationNode annotationNode: metadataNode.annotations()) {
+            if (isGraphqlServiceConfig(annotationNode) && annotationNode.annotValue().isPresent()) {
+                return annotationNode.annotValue().get();
+            }
+        }
+        throw new SchemaGenerationException(DiagnosticMessages.SDL_SCHEMA_102, null, MESSAGE_MISSING_SERVICE_CONFIG);
+    }
+
+    /**
+     * Get schema string field from the given node.
+     */
+    private static String getSchemaStringFieldFromValue(MappingConstructorExpressionNode annotationValue)
+            throws SchemaGenerationException {
+        SeparatedNodeList<MappingFieldNode> existingFields = annotationValue.fields();
+        for (MappingFieldNode field : existingFields) {
+            if (field.children().get(0).toString().contains(SCHEMA_STRING_FIELD)) {
+                String schemaString = field.children().get(2).toString();
+                return schemaString.substring(1, schemaString.length() - 1);
+            }
+        }
+        throw new SchemaGenerationException(DiagnosticMessages.SDL_SCHEMA_102, null,
+                MESSAGE_MISSING_FIELD_SCHEMA_STRING);
+    }
+
+    /**
+     * Check whether the given annotation is a GraphQL service config.
+     *
+     * @param annotationNode     annotation node
+     */
+    private static boolean isGraphqlServiceConfig(AnnotationNode annotationNode) {
```

Comment:
Please check all the places where we have duplicated the code, and if possible, move them to the GraphQL commons. This will cause maintainability issues otherwise.

Reply:
The Exception name is `SchemaGeneartionException`. The name suggests that it should be placed where the schema is generated, not the place where the file is generated.

Reply:
Renamed the exception to `SchemaFileGenerationException`. Fixed in https://github.com/ballerina-platform/graphql-tools/pull/118/commits/152c46cc2b96601de25b9d2d7acae59f647699b6